main.rb

```diff
@@ -0,0 +1,141 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = to_a
+    i = 0
+    while i < array.length
+      yield(array[i], i)
+      i += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |elem| new_array.push(elem) if yield(elem) == true }
+    new_array
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == false }
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    my_each { |elem| return true if yield(elem) == true }
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    my_each { |elem| return false if yield(elem) == true }
+    true
+  end
+
+  def my_count(num = nil)
+    counter = 0
+    if num
+      my_each { |elem| counter += 1 if elem == num }
+
+    elsif block_given?
+      my_each { |elem| counter += 1 if yield(elem) }
+
+    else
+      counter = size
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    return to_enum(:my_map) unless block_given?
+
+    new_array = []
+    if proc
+      my_each { |i| new_array.push(proc.call(i)) }
+    else
+      my_each { |i| new_array.push(yield(i)) }
+    end
+    new_array
+  end
+
+  def my_inject(num = nil, symbol = nil)
+    if block_given?
+      acc = num
+      my_each { |i| acc = acc.nil? ? i : yield(acc, i) }
+      acc
+    elsif acc.nil?
+      !num.nil? && num.is_a?(Symbol) || num.is_a?(String)
+      my_each { |i| acc = acc.nil? ? i : acc.send(num, i) }
+    elsif (acc = num)
+      !num.symbol? && num.is_a?(Symbol) || num.is_a?(String)
+      my_each { |i| acc = acc.nil? ? i : acc.send(symbol, i) }
+    end
+    acc
+  end
```

Comment:
Enumerable `my_inject` raises a "LocalJumpError" when no block or argument is given
      Failure/Error: `expect { array.my_inject }.to raise_error(LocalJumpError)`

        expected LocalJumpError, got #<TypeError: nil is not a symbol nor a string> with backtrace:
##
Enumerable `my_inject` when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
      Failure/Error: `expect(array.my_inject(20, :*)).to eq array.inject(20, :*)`

      TypeError:
        20 is not a symbol nor a string
##
Enumerable `my_inject` when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range
      Failure/Error: `expect(range.my_inject(2, :*)).to eq range.inject(2, :*)`

      TypeError:
        2 is not a symbol nor a string
##